build/shaka-lab.yaml

```diff
@@ -0,0 +1,163 @@
+    # already takes care of that for us (except on Android).
+    - "--unsafely-allow-protected-media-identifier-for-domain=karma.shakalab.rocks"
+    # Normally, Chrome disallows autoplaying videos in many cases.  Enable it
+    # for testing.
+    - "--autoplay-policy=no-user-gesture-required"
+
+  minimum_chrome_android_args: &minimum_chrome_android_args
+    # There is no way in YAML to natively merge arrays, so we start by
+    # duplicating the flags from minimum_chrome_args above.
+    - "--unsafely-allow-protected-media-identifier-for-domain=karma.shakalab.rocks"
+    - "--autoplay-policy=no-user-gesture-required"
+    # On Android we must set --user-data-dir.  WebDriver does not do it for
+    # us as it does on other platforms.  Without --user-data-dir,
+    # --unsafely-allow... does not work.
+    - "--user-data-dir=/data/data/com.android.chrome/cache"
+
+  minimum_chromeos_args: &minimum_chromeos_args
+    # There is no way in YAML to natively merge arrays, so we start by
+    # duplicating the flags from minimum_chrome_args above.
+    - "--unsafely-allow-protected-media-identifier-for-domain=karma.shakalab.rocks"
+    - "--autoplay-policy=no-user-gesture-required"
+    # Allow remote attestation even though the device may be in dev mode.  This
+    # is critical for testing involving L1 content licenses.
+    - "--allow-ra-in-dev-mode"
+
+  chrome_config: &chrome_config
+    goog:chromeOptions:
+      args: *minimum_chrome_args
+
+      # Instruct chromedriver not to disable component updater. The
+      # component updater must run in order for the Widevine CDM to be
+      # available when using a new user-data-dir.
+      # TODO(http://crbug.com/613581): Remove once Chrome bug is fixed.
+      excludeSwitches:
+        - "disable-component-update"
+
+  chrome_android_config: &chrome_android_config
+    goog:chromeOptions:
+      args: *minimum_chrome_android_args
+
+      # Once the new session request reaches chromedriver, it will take
+      # the androidPackage option as a request to start Chrome through
+      # adb on the tethered device.
+      androidPackage: com.android.chrome
+
+  chromeos_config: &chromeos_config
+    # Pass these client-specified arguments through generic-webdriver-server.
+    # This array will be appended after "--" instead of being set through one
+    # specific flag.  For example, with ["--foo", "--bar=baz"] in the args,
+    # this would generate a command like:
+    #   chromeos-webdriver-server.js -- --foo --bar=baz
+    # Those parameters will be passed on to Chrome instead of used by the
+    # WebDriver server.
+    generic:args: *minimum_chromeos_args
+
+
+ChromeMac:
+  browser: chrome
+  os: Mac
+  extra_config: *chrome_config
+
+FirefoxMac:
+  browser: firefox
+  os: Mac
+  extra_config: *firefox_config
+
+Safari:
+  browser: safari
+  os: Mac
+
+SafariTP:
+  # TODO(b/152646297): Safari TP not launching as of Safari 15.4
```

Comment:
Hm. Now that this lab code is being made public, should this bug also be put onto the GitHub bug tracker?

Reply:
I plan to shut down our component in the internal bug-tracker, so this will migrate eventually if it's needed.